lib/jellyfish/config_reader.ex

```diff
@@ -48,19 +50,55 @@ defmodule Jellyfish.ConfigReader do
     end
   end
 
-  def read_nodes(env) do
-    value = System.get_env(env)
-
-    if value not in ["", nil] do
-      value
-      |> String.split(" ", trim: true)
-      |> Enum.map(&String.to_atom(&1))
-    end
-  end
-
   def read_boolean(env) do
     if value = System.get_env(env) do
       String.downcase(value) not in ["false", "f", "0"]
     end
   end
+
+  def read_dist_config() do
+    if read_boolean("JF_DIST_ENABLED") do
+      node_name_value = System.get_env("JF_NODE_NAME")
+      cookie_value = System.get_env("JF_COOKIE")
+      nodes_value = System.get_env("JF_NODES") || ""
+
+      unset_var =
+        [{"JF_NODE_NAME", node_name_value}, {"JF_COOKIE", cookie_value}]
+        |> Enum.find(fn {_env_name, env_val} -> is_nil(env_val) end)
+
+      if unset_var do
+        {unset_var_name, _} = unset_var
+
+        raise """
+        JF_DIST_ENABLED has been set but #{unset_var_name} remains unset.
+        JF_DIST_ENABLED requires following env vars to be also set: JF_NODE_NAME JF_COOKIE.
+        """
+      end
+
+      node_name = parse_node_name(node_name_value)
+      cookie = parse_cookie(cookie_value)
+      nodes = parse_nodes(nodes_value)
+
+      if nodes == [] do
+        Logger.warning("""
+        JF_DIST_ENABLED has been set but JF_NODES remains unset.
+        Make sure that at least one of your Jellyfish instances
+        has JF_NODES set.
+        """)
+      end
```

Comment:
I'm not saying we should raise here, but is there a case where we'd want to leave JF_NODES empty in a Jellyfish meant to be run in a cluster, while other Jellyfishes in said cluster have JF_NODES set 'correctly'?

Reply:
That's correct setup. In fact, first Jellyfish very often won't be able to establish connection to other JFs as they are not alive yet